lib/msf/core/post/linux/kernel.rb

```diff
@@ -0,0 +1,69 @@
+# -*- coding: binary -*-
+require 'msf/core/post/common'
+
+module Msf
+class Post
+module Linux
+module Kernel
+  include ::Msf::Post::Common
+
+  #
+  # Returns the kernel version
+  #
+  # @return [String]
+  #
+  def kernel_version
+    cmd_exec('uname -r').to_s
```

Comment:
Hmm. I suspect that there are plenty of other `uname` variants that might be useful.  I'll PR something to add them.

Reply:
Thanks. `uname -m` comes to mind.

Best to keep non-kernel related properties, such as `uname -n`, out of this mixin, and use the `Msf::Post::Linux::System` mixin instead.